lib/message.ml

```diff
@@ -0,0 +1,17 @@
+open Bin_prot.Std
+
+type category =
```

Comment:
I think we should only go for 
```ocaml
type category =
  | Request
  | Response
  | Failure_detector (* not sure about this one *)
  | Unsupported
[@@deriving bin_io]
```
what do you think?

Reply:
Deku will be our first/main client. Hence, its scope/needs must be the only one we target for the beginning.
>[...] They will be forced to give up on using our library

I don't agree with that, they can still open an issue and ask us to handle a new `category`, don't you think?